src/app/modules/shoppingCart/services/shopping-cart.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+import { take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ShoppingCartService {
+  constructor(private db: AngularFireDatabase) {}
+
+  async AddToCart(course) {
+    const cartId = localStorage.getItem('cartId');
+    if (!cartId) {
+      const cart = await this.db.list('/shoppingCart').push({
+        dateCreated: new Date().getTime()
+      });
+      localStorage.setItem('cartId', cart.key);
+      this.AddCourseCart(cart.key, course);
+    } else {
+      this.AddCourseCart(localStorage.getItem('cartId'), course);
+    }
+  }
+
+  AddCourseCart(idCart, courseAdd) {
+    this.db
+      .object('/shoppingCart/' + idCart + '/items/' + courseAdd.key)
```

Comment:
`/shoppingCard/${idCard}/items/${courseAdd.key}` using string interpolation. What do you think?

Reply:
cart* but yes, that is better